vending-machine/src/main/java/com/sandyhandle/webapp/vendingmachine/controller/VendingMachineController.java

```diff
@@ -1,26 +1,42 @@
 package com.sandyhandle.webapp.vendingmachine.controller;
 
-import com.sandyhandle.webapp.vendingmachine.models.db.Product;
+import com.sandyhandle.webapp.vendingmachine.models.ProductAddRequest;
 import com.sandyhandle.webapp.vendingmachine.models.db.VendingMachine;
 import com.sandyhandle.webapp.vendingmachine.service.VendingMachineService;
+import lombok.Getter;
 import lombok.RequiredArgsConstructor;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
 @RestController
 @RequestMapping("/api/v1/vending-machines")
 @RequiredArgsConstructor
+@Log4j2
 public class VendingMachineController {
 
-    private final VendingMachineService vendingMachineService;
+    @Autowired
+    private VendingMachineService vendingMachineService;
```

Comment:
This is not going to change. So, you need to declare this as final.

Once you declare this as final, you don't need to put `@Autowired` as you already have `@RequiredArgsContructor`. This will automatically inject that into the constructor.